test/unit/create-volume-controller-test.js

```diff
@@ -0,0 +1,121 @@
+describe("UNIT CONTROLLER - createVolumeController", function () {
+    "use strict";
+
+    // Angular injectables
+    var $q, $injector, $controller, $rootScope, $state, $location, $timeout, $templateCache, $compile;
+
+    // Module defined (non-Angular) injectables
+    var $scope, utils, config, createVolume;
+
+    // Local variables used for testing
+    var getAttributeListDeferred, vm;
+
+    // Initialize modules
+    beforeEach(function () {
+        module("TendrlModule");
+        module("templates");
+        module("TestDataModule");
+    });
+
+    // Define all injectables and compile template
+    beforeEach(function () {
+
+        var templateHtml;
+
+        inject(function (_$q_, _$controller_, _$rootScope_, _$state_, _$templateCache_, _$compile_, _createVolume_, _$location_) {
+            $q = _$q_;
+            $controller = _$controller_;
+            $rootScope = _$rootScope_;
+            $location = _$location_;
+            $state = _$state_;
+            $templateCache = _$templateCache_;
+            $compile = _$compile_;
+            createVolume = _createVolume_;
+
+            $scope = $rootScope.$new();
+            templateHtml = $templateCache.get("/modules/volume/create-volume/create-volume.html");
+
+            $compile(templateHtml)($scope);
+        });
+
+        inject(function (_utils_, _config_) {
+            utils = _utils_;
+            config = _config_;
+        });
+    });
+
+    // Create controller instance to be used by all the unit tests
+    beforeEach(function () {
+
+        getAttributeListDeferred = $q.defer();
+
+        sinon.stub(utils, "getAttributeList").returns(getAttributeListDeferred.promise);
+        config.clusterId = "3969b68f-e927-45da-84d6-004c67974f07";
+
+        goTo("create-volume");
+        $state.current.name = "create-volume";
+
+        // Exercise SUT
+        vm = $controller("createVolumeController", {
+            $scope: $scope
+        });
+
+    });
+
+    describe("Initialization Block", function () {
+
+        beforeEach(function () {
+            vm.postUrl = createVolume.actionDetails;
```

Comment:
Its not necessary that Stub will create promise. It can return response as well. 
Eg - sinon.stub(utils, "getActionDetails").returns(response).